sqlobject/src/main/java/org/jdbi/v3/sqlobject/TransactionalHelper.java

```diff
@@ -26,13 +27,13 @@
     {
         try {
             Map<Method, Handler> h = new HashMap<>();
-            h.put(Transactional.class.getMethod("begin"), new BeginHandler());
-            h.put(Transactional.class.getMethod("commit"), new CommitHandler());
-            h.put(Transactional.class.getMethod("rollback"), new RollbackHandler());
+            h.put(Transactional.class.getMethod("begin"), guard(new BeginHandler()));
+            h.put(Transactional.class.getMethod("commit"), guard(new CommitHandler()));
+            h.put(Transactional.class.getMethod("rollback"), guard(new RollbackHandler()));
 
-            h.put(Transactional.class.getMethod("savepoint", String.class), new SavepointHandler());
-            h.put(Transactional.class.getMethod("releaseSavepoint", String.class), new ReleaseSavepointHandler());
-            h.put(Transactional.class.getMethod("rollbackToSavepoint", String.class), new RollbackSavepointHandler());
+            h.put(Transactional.class.getMethod("savepoint", String.class), guard(new SavepointHandler()));
+            h.put(Transactional.class.getMethod("releaseSavepoint", String.class), guard(new ReleaseSavepointHandler()));
+            h.put(Transactional.class.getMethod("rollbackToSavepoint", String.class), guard(new RollbackSavepointHandler()));
```

Comment:
The `rollback`, `savepoint`, `releaseSavepoint`, and `rollbackToSavepoint` methods should all be valid within an `inTransaction` or `useTransaction` call.